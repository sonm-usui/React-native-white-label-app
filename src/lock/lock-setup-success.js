// @flow
import React, { Component } from 'react'
import { StyleSheet, View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import type { Store } from '../store/type-store'
import type { LockSetupSuccessProps, LockSetupSuccessState } from './type-lock'

import { Container, CustomText, CustomButton } from '../components'
import {
  settingsDrawerRoute,
  lockSetupSuccessRoute,
  homeRoute,
} from '../common'
import { unlockApp, clearPendingRedirect } from './lock-store'
import { OFFSET_1X, OFFSET_2X, OFFSET_4X, colors } from '../common/styles'
import { UNLOCKING_APP_WAIT_MESSAGE } from '../common/message-constants'

import { EvaIcon, LOCK_ICON } from '../common/icons'
import { moderateScale } from 'react-native-size-matters'
import { appName } from '../external-imports'

export class LockSetupSuccess extends Component<
  LockSetupSuccessProps,
  LockSetupSuccessState
> {
  onClose = () => {
    this.props.unlockApp()
    if (
      this.props.route &&
      this.props.route.params &&
      this.props.route.params.changePin === true
    ) {
      this.props.navigation.navigate(homeRoute, { screen: settingsDrawerRoute })
    } else if (this.props.pendingRedirection) {
      // if there is a redirection pending, then redirect and clear it
      this.props.pendingRedirection.map((pendingRedirection) => {
        this.props.navigation.navigate(
          pendingRedirection.routeName,
          pendingRedirection.params || {}
        )
      })
      this.props.clearPendingRedirect()
    } else {
      this.props.navigation.navigate(homeRoute)
    }
  }

  render() {
    const { isFetchingInvitation } = this.props
    let message =
      this.props.route &&
      this.props.route.params &&
      this.props.route.params.changePin === true
        ? `Your ${appName} app is secured.`
        : `Your ${appName} app is now secured`

    if (isFetchingInvitation === true) {
      message = UNLOCKING_APP_WAIT_MESSAGE
    }

    return (
      <Container tertiary safeArea>
        <Container clearBg center style={[style.successContainer]}>
          <View style={style.iconBorder}>
            <EvaIcon
              name={LOCK_ICON}
              width={moderateScale(40)}
              height={moderateScale(40)}
              color={colors.main}
            />
          </View>
          <CustomText
            h4
            bg="tertiary"
            tertiary
            thick
            center
            style={[style.successMessage]}
          >
            {message}
          </CustomText>
          <CustomText
            h6
            bg="tertiary"
            tertiary
            thick
            center
            style={[style.successMessage]}
          >
            {this.props.route &&
            this.props.route.params &&
            this.props.route.params.changePin
              ? "From now on you'll need to use your passcode to unlock this app."
              : ' '}
          </CustomText>
        </Container>
        <View style={style.buttonContainer}>
          <CustomButton
            primary
            raised
            medium
            disabled={isFetchingInvitation}
            testID="close-button"
            fontWeight="600"
            title="Close"
            onPress={this.onClose}
          />
        </View>
      </Container>
    )
  }
}

const mapStateToProps = (state: Store, props) => ({
  pendingRedirection: state.lock.pendingRedirection,
  pendingRedirectionParams: state.lock.pendingRedirectionParams || {},
  isFetchingInvitation: Object.keys(state.smsPendingInvitation).some(
    (smsToken) =>
      state.smsPendingInvitation[smsToken] &&
      state.smsPendingInvitation[smsToken].isFetching === true
  ),
  changePin:
    props.route.params !== undefined ? props.route.params.changePin : false,
})

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      clearPendingRedirect,
      unlockApp,
    },
    dispatch
  )

export const lockSetupSuccessScreen = {
  routeName: lockSetupSuccessRoute,
  screen: connect(mapStateToProps, mapDispatchToProps)(LockSetupSuccess),
}

const style = StyleSheet.create({
  iconBorder: {
    width: moderateScale(72),
    height: moderateScale(72),
    borderWidth: 2,
    borderColor: colors.main,
    borderRadius: moderateScale(36),
    alignItems: 'center',
    justifyContent: 'center',
  },
  successContainer: {
    paddingHorizontal: OFFSET_2X,
  },
  successMessage: {
    paddingVertical: OFFSET_4X,
  },
  successInfo: {
    paddingHorizontal: OFFSET_1X,
  },
  buttonContainer: {
    marginHorizontal: '2%',
  },
})
