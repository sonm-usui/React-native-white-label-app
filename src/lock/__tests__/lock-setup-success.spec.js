// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { LockSetupSuccess } from '../lock-setup-success'
import {
  getNavigation,
  pendingRedirection,
} from '../../../__mocks__/static-data'

describe('<LockSetupSuccess />', () => {
  const getProps = (isFetchingInvitation = false) => ({
    isFetchingInvitation,
    navigation: getNavigation(),
    route: {},
    pendingRedirection,
    unlockApp: jest.fn(),
    clearPendingRedirect: jest.fn(),
  })
  const options = {
    createNodeMock: () => {
      return {
        clear: () => {
          // eslint-disable-next-line no-unused-vars
          cleared = true
        },
      }
    },
  }

  let component
  let props
  let cleared

  it('should render Lock setup success Screen', () => {
    props = getProps(false)
    component = renderer.create(<LockSetupSuccess {...props} />, options)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should display Loading while fetching the invitation', () => {
    props = getProps(true)
    component = renderer.create(<LockSetupSuccess {...props} />, options)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
