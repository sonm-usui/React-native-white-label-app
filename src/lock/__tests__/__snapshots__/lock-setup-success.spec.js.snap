// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LockSetupSuccess /> should display Loading while fetching the invitation 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    center={true}
    clearBg={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#86B93B",
          "borderRadius": 34.457142857142856,
          "borderWidth": 2,
          "height": 68.91428571428571,
          "justifyContent": "center",
          "width": 68.91428571428571,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={38.285714285714285}
          bbWidth={38.285714285714285}
          color={4287019323}
          fill="#86B93B"
          focusable={false}
          height={38.285714285714285}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 38,
                "width": 38,
              },
            ]
          }
          tintColor={4287019323}
          vbHeight={24}
          vbWidth={24}
          width={38.285714285714285}
        >
          <RNSVGGroup
            fill={4287019323}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGCircle
                  cx="12"
                  cy="15"
                  r="1"
                />
                <RNSVGPath
                  d="M17 8h-1V6.11a4 4 0 1 0-8 0V8H7a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3v-8a3 3 0 0 0-3-3zm-7-1.89A2.06 2.06 0 0 1 12 4a2.06 2.06 0 0 1 2 2.11V8h-4zM12 18a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontSize": 19,
          },
          Object {
            "fontFamily": "System",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "800",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "paddingVertical": 40,
          },
        ]
      }
      suppressHighlighting={true}
    >
      Loading please wait...
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontSize": 15,
          },
          Object {
            "fontFamily": "System",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "800",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "paddingVertical": 40,
          },
        ]
      }
      suppressHighlighting={true}
    >
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": "2%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "marginLeft": 0,
          "marginRight": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityLabel="close-button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="close-button"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "color": "#ffffff",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 0.4,
              "padding": 8,
              "paddingVertical": 17,
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(208, 8%, 63%)",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<LockSetupSuccess /> should render Lock setup success Screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    center={true}
    clearBg={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#86B93B",
          "borderRadius": 34.457142857142856,
          "borderWidth": 2,
          "height": 68.91428571428571,
          "justifyContent": "center",
          "width": 68.91428571428571,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={38.285714285714285}
          bbWidth={38.285714285714285}
          color={4287019323}
          fill="#86B93B"
          focusable={false}
          height={38.285714285714285}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 38,
                "width": 38,
              },
            ]
          }
          tintColor={4287019323}
          vbHeight={24}
          vbWidth={24}
          width={38.285714285714285}
        >
          <RNSVGGroup
            fill={4287019323}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGCircle
                  cx="12"
                  cy="15"
                  r="1"
                />
                <RNSVGPath
                  d="M17 8h-1V6.11a4 4 0 1 0-8 0V8H7a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3v-8a3 3 0 0 0-3-3zm-7-1.89A2.06 2.06 0 0 1 12 4a2.06 2.06 0 0 1 2 2.11V8h-4zM12 18a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontSize": 19,
          },
          Object {
            "fontFamily": "System",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "800",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "paddingVertical": 40,
          },
        ]
      }
      suppressHighlighting={true}
    >
      Your AppName app is now secured
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontSize": 15,
          },
          Object {
            "fontFamily": "System",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "800",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "paddingVertical": 40,
          },
        ]
      }
      suppressHighlighting={true}
    >
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": "2%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 3,
          "marginLeft": 0,
          "marginRight": 0,
          "overflow": "visible",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityLabel="close-button"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="close-button"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#777777",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "color": "#ffffff",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingVertical": 17,
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
